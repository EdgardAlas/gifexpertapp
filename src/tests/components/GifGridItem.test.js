import React from "react"
import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"

describe("Pruebas en gifgriditem", () => {
    const title = "titulo jein";
    const url = "http://localhost/img.jpg"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    test("debe de mostrar el componente correctamente", () => {


        expect(wrapper).toMatchSnapshot()
    })


    test('debe de tener un parrao con el titulo', () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title)

    })

    test('debe de la imagen igual al url y alt de los props', () => {
        const img = wrapper.find("img");
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })

    test('debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find("div");

        expect(div.hasClass("animate__fadeIn")).toBe(true)

    })



})