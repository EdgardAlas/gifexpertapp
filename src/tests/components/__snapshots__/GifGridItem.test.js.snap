// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifgriditem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo jein"
    src="http://localhost/img.jpg"
  />
  <p>
    titulo jein
  </p>
</div>
`;
